Userland/DevTools/HackStudio/LanguageServers/Cpp/Tests.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2021, Itamar S. <itamar8910@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Tests.h"
+#include "../FileDB.h"
+#include "ParserAutoComplete.h"
+
+using namespace LanguageServers;
+using namespace LanguageServers::Cpp;
+
+static bool s_some_test_failed = false;
+
+#define I_TEST(name)                     \
+    {                                    \
+        printf("Testing " #name "... "); \
+        fflush(stdout);                  \
+    }
+
+#define PASS              \
+    do {                  \
+        printf("PASS\n"); \
+        fflush(stdout);   \
+        return;           \
+    } while (0)
+
+#define FAIL(reason)                   \
+    do {                               \
+        printf("FAIL: " #reason "\n"); \
+        s_some_test_failed = true;     \
+        return;                        \
+    } while (0)
+
+static void test_complete_local_args();
+static void test_complete_local_vars();
+static void test_complete_type();
+static void test_find_variable_definition();
+
+int run_tests()
+{
+    test_complete_local_args();
+    test_complete_local_vars();
+    test_complete_type();
+    test_find_variable_definition();
+    return s_some_test_failed ? 1 : 0;
+}
+
+void test_complete_local_args()
+{
+    I_TEST(Complete Local Args)
+    FileDB filedb;
+    String content = "int main(int argc, char** argv){\n"
+                     "ar\n"
+                     "}\n";
```

Comment:
I think this would be a great use-case for raw string literals :)

Reply:
Huh, cool. TIL about this feature :+1: